tests/test_packages/test_skills/test_generic_buyer/test_behaviours.py

```diff
@@ -0,0 +1,381 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains test case classes based on pytest for AEA end-to-end testing."""
+
+from pathlib import Path
+from typing import cast
+
+from aea.helpers.search.models import Constraint, ConstraintType, Description, Query
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.dialogue.base import DialogueLabel
+from aea.test_tools.test_skill import BaseSkillTestCase
+
+from packages.fetchai.connections.ledger.base import CONNECTION_ID as LEDGER_PUBLIC_ID
+from packages.fetchai.protocols.fipa.message import FipaMessage
+from packages.fetchai.protocols.ledger_api.message import LedgerApiMessage
+from packages.fetchai.protocols.oef_search.message import OefSearchMessage
+from packages.fetchai.skills.generic_buyer.behaviours import GenericSearchBehaviour
+from packages.fetchai.skills.generic_buyer.dialogues import FipaDialogue, FipaDialogues
+from packages.fetchai.skills.generic_buyer.handlers import GenericFipaHandler
+from packages.fetchai.skills.generic_buyer.strategy import GenericStrategy
+
+from tests.conftest import ROOT_DIR
+
+
+FETCHAI = "fetchai"
+
+
+class TestSkillBehaviour(BaseSkillTestCase):
+    """Test behaviours of generic buyer."""
+
+    path_to_skill = Path(ROOT_DIR, "packages", "fetchai", "skills", "generic_buyer")
+
+    def test_search_behaviour_setup_is_ledger_tx(self):
+        """Test the setup method of the search behaviour where is_ledger_tx is True."""
+        # setup
+        search_behaviour = cast(
+            GenericSearchBehaviour, self.skill.skill_context.behaviours.search
+        )
+
+        # before
+        assert self.get_quantity_in_outbox() == 0
```

Comment:
drop, unnecessary

Reply:
These line (and the similar ones below) is not strictly needed, but it makes it explicit for the reader what this function is testing, i.e. there should be 0 outgoing message before the test, and 1 after.

Reply:
Done